### PR TITLE
Connected Elderly dashboard to database and changed some things in MealManager and localizations

### PR DESCRIPTION
**Pull request by JuhoP02**

### CODE REVIEW

**Review by Mnpn — APPROVED** [review]
LGTM

**Review by nouradeen — CHANGES_REQUESTED** [review]
- Adding meals from the caregiver side still does not add meals inside a collection inside the elderly account itself. I don't see why right now, you are calling `userCollection.doc(uid).collection('meals')` which is supposed to add the meals to that location. I can see what I can find.
- In the `MealManager.dart`, I see you are trying to access the meals inside the "elderly" account, through a collection inside of the document. But it is returning no data because of the 
```dart
if (!mealSnapshot.hasData)
{...}
```
So my suggestion now is not to merge the code until I find out the problem.

**Comment by JuhoP02** [comment]
Weird, for me it seems to be working correctly. Remember that the meal editor adds meals for 'tomorrow', and the elderly dashboard returns meals for 'today'.

**Comment by nouradeen** [comment]
Can you try to commit and push your code once again?
I'll try to pull it and test it.

**Comment by JuhoP02** [comment]
The elderly dashboard uses getRemainingMealsForDate, which returns the meals for the date, but also checks which meals are after the current time of day. So you might have to change the time too

**Comment by JuhoP02** [comment]
I added an offset to the timing that can be changed. I think Mohammad is working on the counter logic so I have not touched that.

File: lib/pages/elderly/meal_confirmation.dart [hunk 1]
```diff
@@ -190,28 +202,119 @@ class _MealConfirmationState extends State<MealConfirmation> {
                           alignment: Alignment.bottomCenter,
                           child: GestureDetector(
                             onTap: () {},
```
Comment (nouradeen) [inline lib/pages/elderly/meal_confirmation.dart:204]: - Remove `GestureDetector()`, we had this just for testing the miss screens

File: lib/pages/elderly/meal_confirmation.dart [hunk 2]
```diff
@@ -190,28 +202,119 @@ class _MealConfirmationState extends State<MealConfirmation> {
...
+
+                                  if (!snapshot.hasData) {
+                                    return const CircularProgressIndicator();
+                                  }
+
+                                  final List<QueryDocumentSnapshot<MealInfo>>
+                                      mealDocs = snapshot.data!.docs;
+
+                                  if (mealDocs.isNotEmpty) {
+                                    var nextMealData = mealDocs[0].data();
```
Comment (nouradeen) [inline lib/pages/elderly/meal_confirmation.dart:236]: `mealDocs[0]` is not always the next meal right?, if we have already 3 meals in the database, mealDocs[0] will grab the last one based on the time, beacause they are sorted in the database.

File: lib/pages/elderly/meal_confirmation.dart [hunk 3]
```diff
@@ -190,28 +202,119 @@ class _MealConfirmationState extends State<MealConfirmation> {
...
+
+                    if (!snapshot.hasData) {
+                      return const CircularProgressIndicator();
+                    }
+
+                    final List<QueryDocumentSnapshot<MealInfo>> mealDocs =
+                        snapshot.data!.docs;
+
+                    if (mealDocs.isNotEmpty) {
+                      var nextMealData = mealDocs[0].data();
```
Comment (nouradeen) [inline lib/pages/elderly/meal_confirmation.dart:307]: Same thing here

**Review by nouradeen — CHANGES_REQUESTED** [review]
Fix the comments. I have not tested the code, but I did the same error once before 😃

File: lib/pages/elderly/meal_confirmation.dart (hunk 2 above)
Comment (nouradeen) [reply lib/pages/elderly/meal_confirmation.dart:236]: (First one in the list "in the database")

**Comment by JuhoP02** [comment]
The database should return the meals in ascending order based on the date (lowest/earliest = first in list), so it should return the first one.

**Comment by JuhoP02** [comment]
It is set to mealDocs[0] in meal_confirmation because the 'currentmeal' status gets set from 'Waiting' to 'Done'. Because of that the [0] is the next meal in that screen. Is that what you meant?

**Pull request merged**